### ISSUE TITLE
[System.Text.Json]: Recursive property in IEnumerable type results in stackoverflow

### ISSUE THREAD

**Issue by herrozerro** [dotnet/corefx#40050]
Possibly related to #37313

```
    void Main()
    {
	    testThing thing = new testThing();

	    var s = System.Text.Json.JsonSerializer.Serialize(thing);
    }

    public class testThing
    {
	    public IEnumerable<testThing> things {get;set;}
    }
```

Environment: 3.0.100-preview7-012821

Works with list, but IEnumerable breaks it.

**Comment by ahsonkhan** [comment]
cc @steveharter, @layomia

**Comment by steveharter** [comment]
This is a duplicate of https://github.com/dotnet/corefx/issues/39769 which was just fixed.

I was not able to repro in master with a local build; the fix is [currently being ported to 3.0](https://github.com/dotnet/corefx/pull/40071).
